spec/useless_results_remover_spec.rb

```diff
@@ -32,4 +32,25 @@
     expect(subject).to have_key(source_path)
     expect(subject[source_path]["lines"]).to be_kind_of(Array)
   end
+
+  context "when changing coverage root" do
+    around do |example|
+      begin
+        previous_root_regx = SimpleCov::UselessResultsRemover.instance_variable_get(:@root_regx)
+        SimpleCov::UselessResultsRemover.instance_variable_set(:@root_regx, nil)
+        previous_root = SimpleCov.root
```

Comment:
The spec is very involved and down to nitty gritty impl details. I think it works but overall would feel better if we had a cucumber scenario covering this case. No need to add it here though, I'm fine to have it as is and might add the scenario myself after merge (if I get to it) :)

Reply:
Yep, I agree, super implementation dependent. Another approach is to use a subprocess.

Reply:
Had a super quick look and a cucumber scenario sounds easier and better.

Reply:
yeah I mean that's what cucumber does in our setup, so I'd just add a scenario for it later replacing this. But for getting this shipped and know that it works this is good enough as of now imo so don't feel pressured to change it up :)

Reply:
we should probably rather use a chat :laughing:

Reply:
I can add the feature, no worries! :)

Reply:
I gave the scenario a try, let's see how it goes :crossed_fingers:.